doc/_extensions/zephyr/domain/__init__.py

```diff
@@ -70,6 +69,145 @@
+    "mmu_mpu": nodes.abbreviation(
+        "", "MMU / MPU", explanation="Memory Management Unit / Memory Protection Unit"
+    ),
+    "modem": nodes.Text("Modem"),
+    "mspi": nodes.Text("Multi-bit SPI"),
+    "mtd": nodes.abbreviation("MTD", "MTD", explanation="Memory Technology Device"),
+    "wireless": nodes.Text("Wireless network"),
+    "options": nodes.Text("Options"),
+    "ospi": nodes.Text("Octal SPI"),
+    "pcie": nodes.abbreviation(
+        "PCIe", "PCIe", explanation="Peripheral Component Interconnect Express"
+    ),
+    "peci": nodes.abbreviation(
+        "PECI", "PECI", explanation="Platform Environment Control Interface"
+    ),
+    "phy": nodes.Text("PHY"),
+    "pinctrl": nodes.Text("Pin control"),
+    "pm_cpu_ops": nodes.Text("Power management CPU operations"),
+    "power": nodes.Text("Power management"),
+    "power-domain": nodes.Text("Power domain"),
+    "ppc": nodes.Text("PPC architecture"),
+    "ps2": nodes.abbreviation("PS/2", "PS/2", explanation="Personal System/2"),
+    "pwm": nodes.abbreviation("PWM", "PWM", explanation="Pulse Width Modulation"),
+    "qspi": nodes.Text("Quad SPI"),
+    "regulator": nodes.Text("Regulator"),
+    "reserved-memory": nodes.Text("Reserved memory"),
+    "reset": nodes.Text("Reset controller"),
+    "retained_mem": nodes.Text("Retained memory"),
+    "retention": nodes.Text("Retention"),
+    "riscv": nodes.Text("RISC-V architecture"),
+    "rng": nodes.abbreviation("RNG", "RNG", explanation="Random Number Generator"),
+    "rtc": nodes.abbreviation("RTC", "RTC", explanation="Real Time Clock"),
+    "sd": nodes.Text("SD"),
+    "sdhc": nodes.Text("SDHC"),
+    "sensor": nodes.Text("Sensors"),
+    "serial": nodes.Text("Serial controller"),
+    "shi": nodes.abbreviation("SHI", "SHI", explanation="Secure Hardware Interface"),
+    "sip_svc": nodes.abbreviation("SIP", "SIP", explanation="Service in Platform"),
+    "smbus": nodes.abbreviation("SMBus", "SMBus", explanation="System Management Bus"),
+    "sound": nodes.Text("Sound"),
+    "spi": nodes.abbreviation("SPI", "SPI", explanation="Serial Peripheral Interface"),
+    "sram": nodes.Text("SRAM"),
+    "stepper": nodes.Text("Stepper"),
+    "syscon": nodes.Text("System controller"),
+    "tach": nodes.Text("Tachometer"),
+    "tcpc": nodes.abbreviation("TCPC", "TCPC", explanation="USB Type-C Port Controller"),
+    "test": nodes.Text("Test"),
+    "timer": nodes.Text("Timer"),
+    "timestamp": nodes.Text("Timestamp"),
+    "usb": nodes.Text("USB"),
+    "usb-c": nodes.Text("USB Type-C"),
+    "uac2": nodes.Text("USB Audio Class 2"),
+    "video": nodes.Text("Video"),
+    "virtualization": nodes.Text("Virtualization"),
+    "w1": nodes.Text("1-Wire"),
+    "watchdog": nodes.Text("Watchdog"),
+    "wifi": nodes.Text("Wi-Fi"),
+    "xen": nodes.Text("Xen"),
+    "xspi": nodes.abbreviation("XSPI", "XSPI", explanation="Expanded Serial Peripheral Interface"),
```

Comment:
This large sorted list with some multiline blocks was the inspiration for #80444

Only check lines that match a regex, for example:
```python
    # zephyr-keep-sorted-start /^    "/
    # ...
    # zephyr-keep-sorted-stop
```

Reply:
Ahah yes, *if* this stays I will definitely flag it with the tags, thanks!
Right now I am actually very much interested in --although of course PR is still in draft, ... but since you're here 😄-- figuring out what makes for a "good" categorization of the supported HW feature. As you may have seen from the current implementation, this is making every directory under dts/bindings/ a category, but having ~120 categories doesn't seem right and there's probably a better way? Maybe we keep the approach but do some addition grouping? Like, dunno, off the top of my head, grouping `fuel-gauge` and `battery` ("Battery mgmt"), or `usb` and `usb-c` (USB), ... together ?

Reply:
Oopf, 120 is a lot, I think grouping would make sense, also to have some better names than for example `mm`.